openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -142,7 +142,9 @@ def update_user_preferences(requesting_user, update, user=None):
         if preference_value is not None:
             try:
                 serializer = serializers[preference_key]
-                serializer.save()
+
+                if serializer.instance is None or serializer.instance.value != serializer.validated_data['value']:
```

Comment:
Might be nice to pull this into a helper function since it's reused below and giving it a nice name would make this more readable.